### PR TITLE
Added a Log Category file and the test

### LINKED ISSUE #25: Create Model for LogCategory
A LogCategory contains a record of all the categories that a user intends to spend on in a money log this is usually created as the budget is getting refined, it will contain the following field 
- id: a system generated id 
- name: name of this category 
- money_log_id: id of the MoneyLog for this category 

a model will be needed to 
- [ ] create a LogCategory 
- [ ] update a LogCategory's details 
- [ ] delete a LogCategory

### PR DESCRIPTION
**Pull request by WanjikuMac**
Fixes #25 
Create a LogCategory

### CODE REVIEW

**Comment by zacck-zz** [comment]
@WanjikuMac  could you reference the issue fixed by this, please?

File: lib/spender/planning/log_category.ex [hunk 1]
```diff
@@ -0,0 +1,33 @@
+defmodule Spender.Planning.LogCategory do
+  @moduledoc """
+  This module holds changeset functions used to work with the LogCategory
+  """
+  use Ecto.Schema
+  import Ecto.Changeset
+
+  alias Spender.{
+    MoneyLogs.Budget
+    }
```
Comment (zacck-zz) [inline lib/spender/planning/log_category.ex]: Could you please consider indenting this correctly?

File: lib/spender/planning/log_category.ex [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  @doc """
+  This changeset function takes in a struct and map containing parameters
+  It proceeds to match the parameters in the the map the schema above
+  """
+  def changeset(logcategory, attrs \\ %{}) do
+    logcategory
+    |> cast(attrs, [:name])
+    |> validate_required([:name])
+    |> validate_length(:name, max: 40)
+
```
Comment (zacck-zz) [inline lib/spender/planning/log_category.ex]: this blank line is unnecessary

File: priv/repo/migrations/20180702172407_add_log_category.exs [hunk 3]
```diff
@@ -0,0 +1,18 @@
+defmodule Spender.Repo.Migrations.AddLogCategory do
+  use Ecto.Migration
+  @moduledoc """
+  This module holds the database schema for the logCategory table
+  we use this table to records the logExpense when we log in an expense
+  """
+  @doc """
```
Comment (zacck-zz) [inline priv/repo/migrations/20180702172407_add_log_category.exs]: Could you consider spacing these better perhaps adding a blank line between these two comments

File: priv/repo/migrations/20180702172407_add_log_category.exs [hunk 4]
```diff
@@ -0,0 +1,18 @@
+defmodule Spender.Repo.Migrations.AddLogCategory do
+  use Ecto.Migration
```
Comment (zacck-zz) [inline priv/repo/migrations/20180702172407_add_log_category.exs:6]: Consider putting this below the moduledoc please

File: priv/repo/migrations/20180702172407_add_log_category.exs [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+  @moduledoc """
+  This module holds the database schema for the logCategory table
+  we use this table to records the logExpense when we log in an expense
+  """
+  @doc """
+    This function creates a table with the logCategory details
+  """
+
```
Comment (zacck-zz) [inline priv/repo/migrations/20180702172407_add_log_category.exs]: This space is also unnecessary

File: test/lib/spender/planning/log_category_test.exs [hunk 6]
```diff
@@ -0,0 +1,32 @@
+defmodule Spender.Planning.LogCategoryTest do
+  @moduledoc """
+  This module performs tests for the log-category changeset
+"""
```
Comment (zacck-zz) [inline test/lib/spender/planning/log_category_test.exs]: Please consider indenting this correctly

**Review by zacck-zz — CHANGES_REQUESTED** [review]
@WanjikuMac  please consider using the `mix format` command to lint your files after you have edited them.
Also, in future consider using small commits for each file change rather than one for a big change spanning multiple files

File: lib/spender/planning/log_category.ex (hunk 1 above)
Comment (WanjikuMac) [reply lib/spender/planning/log_category.ex]: Okay

File: lib/spender/planning/log_category.ex [hunk 7]
```diff
@@ -1,20 +1,19 @@
+@moduledoc """
```
Comment (zacck-zz) [inline lib/spender/planning/log_category.ex]: Hey @WanjikuMac  this piece goes inside the module definiton

**Comment by zacck-zz** [comment]
@WanjikuMac  Thank you so much for your work

**Pull request merged**